docker/volumes/db/init/01-auth-schema.sql

```diff
@@ -88,21 +88,57 @@ VALUES  ('20171026211738'),
         ('20180108183307'),
         ('20180119214651'),
         ('20180125194653');
-		
--- Gets the User ID from the request cookie
-create or replace function auth.uid() returns uuid as $$
-  select nullif(current_setting('request.jwt.claim.sub', true), '')::uuid;
-$$ language sql stable;
-
--- Gets the User ID from the request cookie
-create or replace function auth.role() returns text as $$
-  select nullif(current_setting('request.jwt.claim.role', true), '')::text;
-$$ language sql stable;
-
--- Gets the User email
-create or replace function auth.email() returns text as $$
-  select nullif(current_setting('request.jwt.claim.email', true), '')::text;
-$$ language sql stable;
+
+create or replace function auth.uid() 
+returns uuid 
+language plpgsql stable
+as $$
+declare
+    sub text := current_setting('request.jwt.claim.sub', true);
+    claims text := current_setting('request.jwt.claims', true);
+    id uuid;
+begin
+  select case 
+  	when sub is not null then sub::uuid
+  	else (claims::json ->> 'sub')::uuid
+  end into id;
+ 
+  return id;
+end $$;
```

Comment:
New auth functions that can cope with both GUC strategies

Reply:
Since these are extremely hot functions we'll probably want to keep them as sql vs pgsql. SQL functions can optionally be in-lined where plpgsql functions can not.

Intuitively, it shouldn't matter much because `stable` functions should only be executed once per query execution, but in practice postgres doesn't take advantage of that hint as often as it could.

Here's a benchmark on the `uid` functions showing the diff on a 10k record set.

```sql
create or replace function auth.uid_pgsql() 
returns uuid 
language plpgsql stable
as $$
declare
    sub text := current_setting('request.jwt.claim.sub', true);
    claims text := current_setting('request.jwt.claims', true);
    id uuid;
begin
  select case 
  	when sub is not null then sub::uuid
  	else (claims::json ->> 'sub')::uuid
  end into id;

  return id;
end $$;

-- Benchmark
explain analyze
select
    auth.uid_pgsql()
from 
    generate_series(1,10000)

/*
"Function Scan on generate_series  (cost=0.00..2600.00 rows=10000 width=16) (actual time=1.976..107.524 rows=10000 loops=1)"
"Planning Time: 0.041 ms"
"Execution Time: 108.827 ms"
*/
```



vs 

```sql
create or replace function auth.uid_sql() 
returns uuid 
language sql stable
as $$
select 
	coalesce(
		current_setting('request.jwt.claim.sub', true),
		(current_setting('request.jwt.claims', true)::jsonb ->> 'sub')
	)::uuid
$$;
explain analyze
select
	auth.uid_sql()
from 
	generate_series(1,10000)
/*
"Function Scan on generate_series  (cost=0.00..275.00 rows=10000 width=16) (actual time=1.911..10.055 rows=10000 loops=1)"
"Planning Time: 0.127 ms"
"Execution Time: 11.083 ms"
*/
```

So 108 ms - 11 ms means we might accidentally introduce 90 ms of overhead if the user's result set is 10k records.

Reply:
These work good. I was thinking of doing it differently, like:

```sql
if (select current_setting('server_version_num')::int >= 140000) then
-- new notation
else
-- old notation
end if;
```

But the `is not null` logic is working good.